### DISCUSSION TITLE
Scrollbars and ability to define width & height in table, Freezing in the first row and column.

### DISCUSSION THREAD

**Discussion by productshiv** [GitbookIO/community#8] [Feature Requests]
### What's your feature/idea?

Add scrollbars and ability to define width & height in table, Imagine a case where I have 10 rows and 10 columns obviously that will outgrow both vertically and horizontally and if the table lets say speaks about functionality you have on a specific platform than for the 9th platform 9th feature you will be in a lost placement where its tougher to read data.

### What's your desired outcome?

Desired out come would be to have a scrollbar both vertical and horizontal (in the table) and the ability to size the table as big a user wants with freezing row and column.

I have written down the PRD in the anything else section for better and indepth explanation.

### What's the impact of this for you?

- [X] Critical
- [ ] High
- [ ] Medium
- [ ] Low

### Anything else?

Title: Table Width and Height Control with Scrollbars

Introduction:

Creating effective documentation is crucial for any organization. It allows users to understand and navigate through the information easily. However, when it comes to tables with a large amount of data, it can be challenging for users to read and understand the information. Our documentation creating tool currently doesn't provide any control over table dimensions and the overflow of data. This can lead to difficulties while trying to read and understand the information.

Objective:

To improve the user experience while reading and understanding tables with a large amount of data, we propose to add the ability to provide width and height of a table and add scrollbars for the overflowing content. This feature will help users to easily navigate and read through a table.

Functional Requirements:

Allow users to provide table width and height:
Users will be able to set the width and height of a table, so the table fits within the dimensions of the screen.

Add Scrollbars for Overflowing Content:
When the table size grows out of the dimensions set by the user, scrollbars will automatically appear, allowing users to scroll through the data.

Freeze first row and column:
To make it easier to understand the data, the first row and column will be frozen, so they are always visible while scrolling.

Compatibility with existing table formatting options:
The new feature will be compatible with existing table formatting options such as font size, color, and background color.

User-friendly interface:
A user-friendly interface will be provided to set the table width, height, and scrollbar options.

Non-functional Requirements:

Performance:
The new feature will not affect the performance of the documentation creating tool.

Compatibility:
The new feature will be compatible with all modern web browsers.

Accessibility:
The new feature will be accessible to users with disabilities.

Conclusion:

With the ability to provide width and height of a table and add scrollbars for the overflowing content, users will be able to navigate and read through tables with a large amount of data easily. This feature will significantly improve the user experience while working with tables in our documentation creating tool. It will also help to make the documentation more effective, as users will be able to quickly find and understand the information they need.

**Comment by taszap** [comment]
Hey @productshiv 

Firstly thank you for such a well formulated request! 
A while ago we shipped full width blocks which aim at improving user experience with the tables (although I appreciate they are not exactly the solution you had in mind). You can also choose to collapse or show the header. 

I am curious if this has improved your experience at all?
